presto-iceberg/src/main/java/com/facebook/presto/iceberg/rest/IcebergRestCatalogFactory.java

```diff
@@ -67,6 +68,8 @@ protected Map<String, String> getCatalogProperties(ConnectorSession session)
 
         catalogConfig.getAuthenticationType().ifPresent(type -> {
             if (type == OAUTH2) {
+                catalogConfig.getAuthenticationServerUri().ifPresent(authServerUri -> properties.put(OAUTH2_SERVER_URI, authServerUri));
+
```

Comment:
Since fallback will be deprecated in the future - https://github.com/apache/iceberg/pull/10603- should we also add a comment about the same? And eventually, we can make this a mandate property in the case of OAUTH2?

Reply:
good idea! Will add shortly